### DISCUSSION TITLE
hdparm service failing to start

### DISCUSSION THREAD

**Discussion by iamaburrama** [retronas/discussions#5] [Q&A] [answered]
Not sure if things were changed around but there used to be an "Advanced" section in the retronas menu that housed a few things like the hdparm utility.  Did this get moved to the Install section.  Not really an issue just wasn't sure if that was it's new location and that the same functions acted the same.

The main issue I'm having is I'm trying to start the hdparm service and have this start anytime the system is rebooted but it looks like the service is failing to start.  Typically after a restart of the server I'd have to go into the hdparm menu and manually start this.  Then I could check the status and all would be good. 

In an attempt to get this to start automatically after reboot I installed the service from the Install menu.  Once complete I went into Cockpit to start this but keep getting errors that the service fails to start.  The errors I see in the Status menu are attached.  

Thank you for any help you can provide.

<img width="741" alt="2023-11-24_09-07-39" src="https://github.com/danmons/retronas/assets/42190058/d6f48b29-f7cc-4d96-87d6-8c8023356e71">

If it makes any difference I'm running retronas on a vm through unraid.

**Comment by sairuk** [comment, marked as answer]
you'll need to run the script manually to work out why its failing in your setup, i don't run unraid, but i took a look and the only exit 1 i don't have feedback on is the failure to find the config file, did you configure a drive?

`/opt/retronas/scripts/hdparm.sh` as root

this file needs to exist with content which should be generated from the menu
`stat /data/retronas/config/hdparm-manager-drive`

**Reply by iamaburrama** [reply]
Oh wow thanks for that.  That tipped me off to what the problem was.  I ran the drive selector and selected my drive.  After that I was able to start the service successfully.

<img width="741" alt="2023-11-24_20-02-07" src="https://github.com/danmons/retronas/assets/42190058/65b0823c-5493-41e7-860f-3a0c507a7fe1">

<img width="741" alt="2023-11-24_20-04-51" src="https://github.com/danmons/retronas/assets/42190058/bd2a60ae-bf48-4f38-abbb-8bdae9e41487">

Thanks so much for your help!

**Reply by sairuk** [reply]
i'll add some feedback in the future for that exit, thanks for the report

**Reply by sairuk** [reply]
we now have this output

```
Nov 25 12:26:51 testronas systemd[1]: Started HDPARM HDD No Sleep.
Nov 25 12:26:51 testronas hdparm.sh[104322]: Failed to find config file
Nov 25 12:26:51 testronas systemd[1]: hdparm.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 12:26:51 testronas systemd[1]: hdparm.service: Failed with result 'exit-code'.
```
